File: src/main/java/um/tesoreria/core/model/MercadoPagoContext.java
package um.tesoreria.core.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.json.JsonMapper;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;
import um.tesoreria.core.kotlin.model.Auditable;
import um.tesoreria.core.util.Jsonifier;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = true)
public class MercadoPagoContext extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long mercadoPagoContextId;
    
    private Long chequeraCuotaId;
    private String initPoint;
    
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime fechaVencimiento;
    
    private BigDecimal importe = BigDecimal.ZERO;
    private Byte changed = 0;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime lastVencimientoUpdated;

    private String preferenceId;
    private String preference;
    private Byte activo = 0;
    private Long chequeraPagoId;
    private String idMercadoPago;
    private String status;
    
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime fechaPago;
    
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime fechaAcreditacion;
    
    private BigDecimal importePagado = BigDecimal.ZERO;
    private String payment;

    public String jsonify() {
        return Jsonifier.builder(this).build();
    }

}

File: src/main/java/um/tesoreria/core/kotlin/model/MercadoPagoContext.kt
package um.tesoreria.core.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType

import java.time.OffsetDateTime
import java.math.BigDecimal

@Entity
data class MercadoPagoContext(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var mercadoPagoContextId: Long? = null,
    var chequeraCuotaId: Long? = null,
    var initPoint: String? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaVencimiento: OffsetDateTime? = null,
    var importe: BigDecimal = BigDecimal.ZERO,
    var preference: String? = null,
    var activo: Byte = 0,
    var chequeraPagoId: Long? = null,
    var idMercadoPago: String? = null,
    var status: String? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaPago: OffsetDateTime? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaAcreditacion: OffsetDateTime? = null,

    var importePagado: BigDecimal = BigDecimal.ZERO,
    var payment: String? = null

) : Auditable() {

    class Builder {
        private var mercadoPagoContextId: Long? = null
        private var chequeraCuotaId: Long? = null
        private var initPoint: String? = null
        private var fechaVencimiento: OffsetDateTime? = null
        private var importe: BigDecimal = BigDecimal.ZERO
        private var preference: String? = null
        private var activo: Byte = 0
        private var chequeraPagoId: Long? = null
        private var idMercadoPago: String? = null
        private var fechaPago: OffsetDateTime? = null
        private var importePagado: BigDecimal = BigDecimal.ZERO
        private var payment: String? = null
        private var status: String? = null
        private var fechaAcreditacion: OffsetDateTime? = null

        fun mercadoPagoContextId(mercadoPagoContextId: Long?) = apply { this.mercadoPagoContextId = mercadoPagoContextId }
        fun chequeraCuotaId(chequeraCuotaId: Long?) = apply { this.chequeraCuotaId = chequeraCuotaId }
        fun initPoint(initPoint: String?) = apply { this.initPoint = initPoint }
        fun fechaVencimiento(fechaVencimiento: OffsetDateTime?) = apply { this.fechaVencimiento = fechaVencimiento }
        fun importe(importe: BigDecimal) = apply { this.importe = importe }
        fun preference(preference: String?) = apply { this.preference = preference }
        fun activo(activo: Byte) = apply { this.activo = activo }
        fun chequeraPagoId(chequeraPagoId: Long?) = apply { this.chequeraPagoId = chequeraPagoId }
        fun idMercadoPago(idMercadoPago: String?) = apply { this.idMercadoPago = idMercadoPago }
        fun fechaPago(fechaPago: OffsetDateTime?) = apply { this.fechaPago = fechaPago }
        fun importePagado(importePagado: BigDecimal) = apply { this.importePagado = importePagado }
        fun payment(payment: String?) = apply { this.payment = payment }
        fun status(status: String?) = apply { this.status = status }
        fun fechaAcreditacion(fechaAcreditacion: OffsetDateTime?) = apply { this.fechaAcreditacion = fechaAcreditacion }

        fun build() = MercadoPagoContext(
            mercadoPagoContextId,
            chequeraCuotaId,
            initPoint,
            fechaVencimiento,
            importe,
            preference,
            activo,
            chequeraPagoId,
            idMercadoPago,
            status,
            fechaPago,
            fechaAcreditacion,
            importePagado,
            payment
        )
    }

    companion object {
        fun builder() = Builder()
    }

}
